idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -1536,3 +1535,199 @@ def sample_points(self):
+            elif len(data_input[0]) != len(data_input[1]):
+                raise Exception("data_input must contain two lists of equal lengths.")
+            elif data_input[0] == data_input[1]:
+                raise Exception("Invalid entry: both lists are equal.")
+            else:
+                bounds_array = np.zeros(
+                    (
+                        2,
+                        len(data_input[0]),
+                    )
+                )
+                bounds_array[0, :] = np.array(data_input[0])
+                bounds_array[1, :] = np.array(data_input[1])
+                data_headers = []
+            self.data = bounds_array
+            self.data_headers = data_headers
+
+            # Catch potential errors in number_of_samples
+            if number_of_samples is None:
+                print(
+                    "\nNo entry for number of samples to be generated. The default value of 5 will be used."
+                )
+                number_of_samples = 5
+            elif not isinstance(number_of_samples, int):
+                raise Exception("number_of_samples must be an integer.")
+            elif number_of_samples <= 0:
+                raise Exception("number_of_samples must a positive, non-zero integer.")
+            self.number_of_samples = number_of_samples
+            self.x_data = bounds_array  # Only x data will be present in this case
+
+        # Check that list_of_distributions is a list, list length is correct and all list values are strings
+        if list_of_distributions is None:
+            raise ValueError("list_of_distributions cannot be empty.")
+        if not isinstance(list_of_distributions, list):
+            raise TypeError("Error with list_of_distributions: list required.")
+        if len(list_of_distributions) != self.x_data.shape[1]:
+            raise ValueError(
+                "Length of list_of_distributions must equal the number of variables."
+            )
+        if all(isinstance(q, str) for q in list_of_distributions) is False:
+            raise TypeError("All values in list must be strings")
+        if not all(
+            q.lower() in ["random", "normal", "uniform"] for q in list_of_distributions
+        ):
+            raise ValueError(
+                "list_of_distributions only supports 'random', 'normal' and 'uniform' sampling options."
+            )
+        self.dist_vector = list_of_distributions
+
+    def generate_from_dist(self, dist_name):
+        if dist_name.lower() in ["uniform", "random"]:
+            dist = getattr(np.random.default_rng(), dist_name.lower())
+            var_values = np.array(dist(size=self.number_of_samples))
+            return dist, var_values
+        elif dist_name.lower() == "normal":
+            dist = getattr(np.random.default_rng(), "normal")
+            var_values = dist(loc=0.5, scale=1 / 6, size=self.number_of_samples)
```

Comment:
How are you determining the mean and standard deviation here? From the code, I suspect you are taking the mean to be the average of the upper and lower bound and the standard deviation to be the difference in bounds divided by 6.

I think most users would expect to provide the mean and standard deviation explicitly, with the possibility of providing upper and lower bounds to prevent extreme values.

Reply:
Resolved - we just need to document this properly.

Reply:
Done now.